docs/bigquery/reference.rst

```diff
@@ -1,68 +1,133 @@
 API Reference
 ~~~~~~~~~~~~~
 
-.. automodule:: google.cloud.bigquery
-  :no-members:
+.. currentmodule:: google.cloud.bigquery
+
+The main concepts with this API are:
+
+- :class:`~google.cloud.bigquery.client.Client` manages connections to the
+  BigQuery API. Use the client methods to run jobs (such as a
+  :class:`~google.cloud.bigquery.job.QueryJob` via
+  :meth:`~google.cloud.bigquery.client.Client.query`) and manage resources.
+
+- :class:`~google.cloud.bigquery.dataset.Dataset` represents a
+  collection of tables.
+
+- :class:`~google.cloud.bigquery.table.Table` represents a single "relation".
 
 Client
 ======
 
-.. automodule:: google.cloud.bigquery.client
-  :members:
-  :show-inheritance:
+.. autosummary::
+    :toctree: generated
 
+    client.Client
 
 Job
 ===
 
-.. automodule:: google.cloud.bigquery.job
-  :members:
-  :show-inheritance:
+Job Configuration
+-----------------
+
+.. autosummary::
+    :toctree: generated
+
+    job.QueryJobConfig
+    job.CopyJobConfig
+    job.LoadJobConfig
+    job.ExtractJobConfig
+
+Job Classes
+-----------
+
+.. autosummary::
+    :toctree: generated
+
+    job.QueryJob
+    job.CopyJob
+    job.LoadJob
+    job.ExtractJob
+    job.UnknownJob
+
+Job Resources
```

Comment:
"Resources" doesn't seem right here:  it suggests things fetchable via an API request.  How about "Job-Related Types" or "Enumerations"?

Reply:
I'll use "Job-Related Types" to keep it open to non-enumerations if we ever add some.